src/components/cylc/tree/cylc-tree.js

```diff
@@ -324,7 +332,9 @@ class CylcTree {
       // the parent-child could end up repeated by accident; it means we must make sure to create this relationship
       // exactly once.
       if (parent.children.length === 0 || !parent.children.find(child => child.id === familyProxy.id)) {
-        parent.children.push(familyProxy)
+        const names = parent.children.map(child => child.node.name)
+        const insertIndex = sortedIndex(names, familyProxy.node.name)
+        parent.children.splice(insertIndex, 0, familyProxy)
```

Comment:
For the other elements, we know we can just add in the next index (i.e. the index where the elements would still be in order).